embassy-rp/src/pwm.rs

```diff
@@ -301,78 +301,78 @@ impl<'d, T: Channel> Drop for Pwm<'d, T> {
 }
 
 mod sealed {
-    pub trait Channel {}
+    pub trait Slice {}
 }
 
-/// PWM Channel.
-pub trait Channel: Peripheral<P = Self> + sealed::Channel + Sized + 'static {
-    /// Channel number.
+/// PWM Slice.
+pub trait Slice: Peripheral<P = Self> + sealed::Slice + Sized + 'static {
+    /// Slice number.
     fn number(&self) -> u8;
 
-    /// Channel register block.
+    /// Slice register block.
     fn regs(&self) -> pac::pwm::Channel {
         pac::PWM.ch(self.number() as _)
     }
 }
 
-macro_rules! channel {
+macro_rules! slice {
     ($name:ident, $num:expr) => {
-        impl sealed::Channel for peripherals::$name {}
-        impl Channel for peripherals::$name {
+        impl sealed::Slice for peripherals::$name {}
+        impl Slice for peripherals::$name {
             fn number(&self) -> u8 {
                 $num
             }
         }
     };
 }
 
-channel!(PWM_SLICE0, 0);
-channel!(PWM_SLICE1, 1);
-channel!(PWM_SLICE2, 2);
-channel!(PWM_SLICE3, 3);
-channel!(PWM_SLICE4, 4);
-channel!(PWM_SLICE5, 5);
-channel!(PWM_SLICE6, 6);
-channel!(PWM_SLICE7, 7);
-
-/// PWM Pin A.
-pub trait PwmPinA<T: Channel>: GpioPin {}
-/// PWM Pin B.
-pub trait PwmPinB<T: Channel>: GpioPin {}
-
-macro_rules! impl_pin {
+slice!(PWM_SLICE0, 0);
+slice!(PWM_SLICE1, 1);
+slice!(PWM_SLICE2, 2);
+slice!(PWM_SLICE3, 3);
+slice!(PWM_SLICE4, 4);
+slice!(PWM_SLICE5, 5);
+slice!(PWM_SLICE6, 6);
+slice!(PWM_SLICE7, 7);
+
+/// PWM Channel A.
+pub trait PwmChannelA<T: Slice>: GpioPin {}
```

Comment:
I'd prefer to keep the pin traits as `PwmPin*`, for consistency with other drivers (like TxPin, SdaPin...). These "pin traits" indicate "this pin is usable for this function", it makes sense to call them all "*Pin".

Reply:
Or perhaps we can name them `ChannelAPin`, `ChannelBPin`. as in "pin usable for channel A". Consistent with "TxPin" as in "Pin usable for TX".

Reply:
I think `ChannelAPin` and `ChannelBPin` work very nice.